wagtail/contrib/settings/context_processors.py

```diff
@@ -67,6 +69,84 @@ def _inner(request):
             # so no settings can be idenfified
             return {}
         else:
-            return SettingsProxy(request)
+            return SiteSettingProxy(request)
+
+    return {"site_settings": SimpleLazyObject(lambda: _inner(request))}
+
+
+def settings(request):
+    # delay site query until settings values are needed
+    def _inner(request):
+        warnings.warn(
```

Comment:
The 'site settings' template tag will default to the first site, and accepts a `is_default_site` value, though the 'generic settings' tag does not and will only ever be for a singleton.

Does it make sense to combine them?

Reply:
Oh wait, if you mean because the warning session refers to `site_settings`, that's because `settings` are currently `site_settings`, no? Generic settings are newwwww.

Reply:
Sorry my comment wasn't very clear!

Would it be possible to make the template tag return both site settings and global settings? I think this would make the API easier to use since you would only need one settings tag for both.

For example, you could define your settings with different base models:

```python

class MyGenericSettingsModel(BaseGenericSetting):
    pass


class MySiteSettingsModel(BaseSiteSetting):
    pass
```

But using them would be the same. The behaviour of which settings model instance is obviously dependant on whether the model is site-specific (or later, locale-specific), but I don't think the template author needs to be concerned about that.

```html
{% load wagtailsettings_tags %}
{% get_settings %}
{{ settings.app_label.MyGenericSettingsModel.x }}
{{ settings.app_label.MySiteSettingsModel.x }}
```

Reply:
Hmm, I'm not sure. The site settings tag can work without a request object (as `is_default_site` can be used instead), whereas the generic settings tag requires a request. They are different things, too. While I see the convenience in a single tag, each model (generic/site) has different functionality.

Reply:
Why does the generic setting tag need a request? Isn't the whole point of it that it's a single instance for all requests across all sites?

Reply:
It's being used for caching, though I'll try to remove that dependency now...

Reply:
`request` now removed: https://github.com/wagtail/wagtail/pull/8327/commits/93645fbd2a47edf6b5ec46d832b66bf4e9dd09fe (along with some caching to save time as something that can be added later).

Reply:
> Though I'm still unsure about combining these two things into a single template tag as the functionality is still quite different for site and generic settings.

Is it possible to implement the API as I described in https://github.com/wagtail/wagtail/pull/8327#discussion_r849302047? If not, please let me know!

I don't think it matters much if the implementation is a little messy, a good developer experience should take priority over a clean implementation in my opinion.